src/main/java/homework_3/ImmutableClass.java

```diff
@@ -0,0 +1,46 @@
+package homework_3;
+
+import java.util.ArrayList;
+import java.util.List;
+/*
+The class must be declared as final
+Data members in the class must be declared as private
+Data members in the class must be declared as final
+A parameterized constructor should initialize all the fields performing a deep copy
+Deep Copy of objects should be performed in the getter methods
+No setters
+ */
+public final class ImmutableClass {
+    private final String aString;
+    private final int anInt;
+    private final List<String> aList;
+
+    public ImmutableClass(List<String> aList) {
+        this.aString = "default";
+        this.anInt = 0;
+        this.aList = aList;
```

Comment:
Actually, I see now your class is mutable. A parameterized constructor should initialize all the fields performing a deep copy

Reply:
done here